### ISSUE TITLE
changing the value of "action" does not change "lockstate"

### ISSUE THREAD

**Issue by philoxio** [Zefau/ioBroker.nuki2#13]
Hi,

when triggering the lock through a new value in "action" the "lockstate" is not change and the command is not transferred to the lock.

![image](https://user-images.githubusercontent.com/50422054/57398867-4cf87b80-71d0-11e9-8e7d-a57aa43cb9fd.png)

Any idea why this is the case?

**Comment by Zefau** [comment]
Thanks for reporting the issue. Could you please change your instance to debug and provide the log ?

**Comment by philoxio** [comment]
I am on V0.9.9 and I have both, Bridge and Web API set up.
`

nuki2.0 | 2019-05-08 21:32:07.813 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
-- | -- | -- | --
nuki2.0 | 2019-05-08 20:54:45.869 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:38:55.003 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:38:24.985 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:35:25.159 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:34:54.938 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:31:54.685 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:28:54.783 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:26:24.905 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:25:54.516 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:24:54.772 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:24:24.578 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:23:54.463 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:22:24.633 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:21:54.377 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:21:24.357 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:20:54.281 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:20:24.361 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:19:54.104 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:18:54.252 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:18:24.527 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:18:05.209 | info | Callback attached to Nuki Haustür.
nuki2.0 | 2019-05-08 20:18:03.966 | info | Listening for Nuki events on port 9076.
nuki2.0 | 2019-05-08 20:18:03.927 | info | starting. Version 0.9.9 in /opt/iobroker/node_modules/iobroker.nuki2, node: v8.15.1
nuki2.0 | 2019-05-08 20:18:03.857 | info | States connected to redis: 127.0.0.1:6379
nuki2.0 | 2019-05-08 20:17:58.171 | info | Adapter stopped und unloaded.
nuki2.0 | 2019-05-08 20:17:58.162 | info | Adapter stopped und unloaded.
nuki2.0 | 2019-05-08 20:13:09.671 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:12:39.676 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:10:09.748 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:09:50.607 | info | Callback attached to Nuki Haustür.
nuki2.0 | 2019-05-08 20:09:49.447 | info | Listening for Nuki events on port 9076.
nuki2.0 | 2019-05-08 20:09:49.437 | info | No Nuki Web API token provided.
nuki2.0 | 2019-05-08 20:09:49.430 | info | starting. Version 0.9.9 in /opt/iobroker/node_modules/iobroker.nuki2, node: v8.15.1
nuki2.0 | 2019-05-08 20:09:49.356 | info | States connected to redis: 127.0.0.1:6379
nuki2.0 | 2019-05-08 20:09:43.638 | info | Adapter stopped und unloaded.
nuki2.0 | 2019-05-08 20:09:43.631 | info | Adapter stopped und unloaded.
nuki2.0 | 2019-05-08 20:08:24.615 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:07:54.501 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:06:54.732 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:06:24.487 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:04:54.527 | warn | Connection settings for bridge incorrect with ID 257086755! No connection established. See debug log for more details.
nuki2.0 | 2019-05-08 20:03:55.376 | info | Callback attached to Nuki Haustür.
nuki2.0 | 2019-05-08 20:03:54.130 | info | Listening for Nuki events on port 9076.
nuki2.0 | 2019-05-08 20:03:54.093 | info | starting. Version 0.9.9 in /opt/iobroker/node_modules/iobroker.nuki2, node: v8.15.1
nuki2.0 | 2019-05-08 20:03:54.039 | info | States connected to redis: 127.0.0.1:6379
nuki2.0 | 2019-05-08 20:03:46.022 | info | Adapter stopped und unloaded.

`

**Comment by Zefau** [comment]
Thanks for the log. I won't be able to fix this until this weekend. Please go back to stable version 0.9.5, which should work fine.

**Comment by philoxio** [comment]
Thanks for the quick replies! Let me know once you´re done I´m happy to test it out.

**Comment by Zefau** [comment]
I will, thank you very much for your support.

**Comment by Zefau** [comment]
Could you please change your instance to `debug` and post the debug details as well?
![debug](https://user-images.githubusercontent.com/14368811/57517987-542aa100-7318-11e9-9f7b-6b486b5ab362.PNG)

**Comment by Zefau** [comment]
Please install current latest release v0.9.10 from Github via SSH:

```
iobroker url https://github.com/Zefau/ioBroker.nuki2
iobroker upload nuki2
```

**Comment by philoxio** [comment]
I installed the v0.9.10 and the problem is gone! Thanks!

Would you still like to have the debug details?

**Comment by Zefau** [comment]
No, that's ok. Thanks for the issue and your support.

**Comment by senger1985** [comment]
I've the same problem, with v1.0.1.
Update in Sec: 20
If I restart the adapter, the state will getting right.

**Comment by Zefau** [comment]
Can you please provide the log including a lock action that has been applied to your lock?
Please also post a screenshot of your current settings as well as the result when calling `http://<bridgeIp>:<bridgePort>/callback/list?token<token>` (https://developer.nuki.io/page/nuki-bridge-http-api-190/4#heading--callback-list).

**Comment by senger1985** [comment]
2019-08-14 11:23:34.064 | debug | Received payload via callback: {"deviceType":0,"nukiId":xxx,"mode":2,"state":3,"stateName":"unlocked","batteryCritical":false}

**Comment by Zefau** [comment]
Port 8080 ist zu niedrig und wahrscheinlich auch besetzt? Wundert mich, dass der Adapter aber trotzdem 51988 nutzt.

Sieht aber sonst gut aus. Der Payload wird ja offensichtlich über den Callback empfangen? Payload sieht auch korrekt aus.
Nach dem Log-Eintrag `Received payload via callback` kommt nichts mehr?

**Comment by senger1985** [comment]
Nope

**Comment by Zefau** [comment]
Ja

**Comment by Zefau** [comment]
Bitte mal die aktuelle Version v1.0.2 von Github installieren. Da wird nun mehr im Debug-Log ausgegeben, um den Fehler einzugrenzen.

**Comment by senger1985** [comment]
nuki2.0 | 2019-08-14 12:45:18.606 | warn | Connection settings for bridge incorrect with ID 1293114678! No connection established. See debug log for more details.
-- | -- | -- | --

**Comment by senger1985** [comment]
Das Objekt "lastAction" wird direkt aktualisiert.
"lockState" dagegen, bleibt, wie es ist.

**Comment by Zefau** [comment]
Bitte schick mir das komplette Log, damit ich den Fehler eingrenzen kann.

**Comment by Zefau** [comment]
https://cryptobin.co/ mit Passwort

**Comment by senger1985** [comment]
Wie kann ich dir hier eine Nachricht schicken?

**Comment by Zefau** [comment]
zefau@mailbox.org

**Comment by senger1985** [comment]
Done

**Comment by Zefau** [comment]
hast du auf v1.0.2 geupdated? 
```
iobroker url https://github.com/Zefau/ioBroker.nuki2
iobroker upload nuki2
```

**Comment by senger1985** [comment]
Ja, habe ich, aber er aktualisiert tatsächlich erst wieder nach dem Neustart, des Adapters.
Bin aber jetzt auch erstmal unterwegs und evtl. erst wieder spät am Rechner.

**Comment by senger1985** [comment]
Frag mich nicht, was sich wie geändert hat, aber heute scheint alles richtig angezeigt zu werden.
Werde es aber die Tage noch weiter beobachten.

Theoretisch könntest du den Fall schließen, wenn du der Meinung bist, es wäre erledigt.

**Comment by senger1985** [comment]
Wie schon geschrieben, bei mir werden die Werte jetzt wirklich angezeigt, außer bei einer Aktion.

Wenn ich die Tür schließe und Nuki automatisch zuschließt, wird das Objekt "lockState" nicht aktualisiert. Nur, wenn ich die Tür via Schlüssel, Knopfdruck, oder App schließe, wird der Wert aktualisiert.

**Comment by Zefau** [comment]
@philoxio @senger1985 I would appreciate if you could verify this issue on the new version v2.0.0 of this adapter. As of now, it is a release candidate and not 100% stable.
See https://github.com/Zefau/ioBroker.nuki2/issues/18#issuecomment-533825482 and https://forum.iobroker.net/post/304268.

Thanks in advanced

**Comment by Zefau** [comment]
Should be fixed with nuki-extended v2.0.0

**Comment by jpawlowski** [comment]
@Zefau wenn ich über MQTT den _ACTION Wert setze, dann wird er zwar in der Admin Oberfläche entsprechend umgeschrieben. Es wird jedoch kein Signal an das Schloss geschickt.
Ist das ein Bug speziell beim nuki-extended Adatper, eher eine MQTT-Client Adapter Geschichte, oder gar ein Bug im js-controller?
